### PR TITLE
Onboarding high fidelity conversion

### PR DESCRIPTION
**Pull request by rs929**
## Overview

Conversion of Scoop onboarding to high fidelity:
Changed Constraints
Added visual features in assets
Added fonts
Created Back Button
Rearranged pageview

## Changes Made

**In OnboardingPageViewController:**
Added functions to implement back button on onboarding screens:
Moved ProfileViewController to the very back of the pageview pages list (according to Figma)
presentPreviousVC()
didTapBack()

**In OnboardingDelegate:**
Added a function to create back button on the navigation bar: setupBackButton()
Added a function to set the visibility of the Back Back Button
Added a function to set the title with proper font to each onboarding view
Created a new class of text fields with the proper insets and outlines: OnboardingTextField

**In all Onboarding View Controllers:**
Replaced normal text fields with OnboardingTextField
Edited the constraints to fit high fidelity
Added tick marks for sliders on PreferencesViewController
Added upload and add photo later buttons to ProfileViewController to replace green button
Adjusted multipliers for the title lines lengths to better fit high-fidelity design
Changed fonts of all labels and textfields to fit high-fidelity
Changed the placeholder text for textfields

## Test Coverage

Ran the simulator on iphones of many different sizes to ensure that it all looked proper
Tested the back button with next button on each screen

## Related PRs or Issues (delete if not applicable)

Addition to green button PR

## Screenshots (delete if not applicable)

Simulator Screen Shot - iPhone 14 - 2022-11-01 at 21 07 21
Simulator Screen Shot - iPhone 14 - 2022-11-01 at 21 07 25
Simulator Screen Shot - iPhone 14 - 2022-11-01 at 21 07 29
Simulator Screen Shot - iPhone 14 - 2022-11-01 at 21 07 32
Simulator Screen Shot - iPhone 14 - 2022-11-01 at 21 07 35
Simulator Screen Shot - iPhone 14 - 2022-11-01 at 21 07 38

### CODE REVIEW

**Review by emarc0314 — APPROVED** [review]
Reviewed changes with Richie in person so LGTM 🚀

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:91]: I don't think you are using a StackView correctly. The idea behind a StackView is that you shouldn't really be setting the position of a subview relative to another subview in that StackView. The StackView itself should be doing all the positioning. The only exception I can think of this is when you are setting size constraints inside a StackView.

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:111]: Same thing here, you shouldn't have to position views relative to each other when they are inside a stackview

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:126]: here too

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:147]: here as well

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [reply Scoop/Controller/Onboarding/AboutYouViewController.swift:91]: To set the spacing to be something different than what you define for the spacing of the stack view you would use `setCustomSpacing` https://developer.apple.com/documentation/uikit/uistackview/2866023-setcustomspacing

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:150]: add newline

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:154]: add newline

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (emarc0314) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:91]: instead of constraining the top anchors to one another within a stackview, use stackView.setCustomSpacing. More [here](https://developer.apple.com/documentation/uikit/uistackview/2866023-setcustomspacing)

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (emarc0314) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:126]: same as previous

File: Scoop/Controller/Onboarding/FavoritesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift]: Same thing with stackview custom spacing here

File: Scoop/Controller/Onboarding/FavoritesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift]: Yeah just fix this for each time you set the position after arranging a subview

File: Scoop/Controller/Onboarding/OnboardingPageViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingPageViewController.swift]: I would guard let this viewControllerIndex

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift:19]: can this be private?

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift]: These are internal by default, so I would remove the internal keyword

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift]: Can this be private?

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift]: You shouldn't be centering the Y and position to the superview. That would result in conflicting constraints. Instead I would choose which one you want to actually pin to the superview

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift]: Delete extra line

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift]: You don't really need this function. You can just do `backButton.isHidden.toggle()` If a button is hidden it is automatically doesn't detect input events so isEnabled logic is not needed.

File: Scoop/Controller/Onboarding/PhoneNumberViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/PhoneNumberViewController.swift]: In this case you would just do `backButton.isHidden = true` without needing the helper function

File: Scoop/Controller/Onboarding/PhoneNumberViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/PhoneNumberViewController.swift]: `leadingTrailingInset`

File: Scoop/Controller/Onboarding/PreferencesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/PreferencesViewController.swift]: `backButton.isHidden = false`

File: Scoop/Controller/Onboarding/PreferencesViewController.swift
Comment (ReadeP) [inline Scoop/Controller/Onboarding/PreferencesViewController.swift:176]: nit: `musicSlider.maximumValue / 2`

**Review by ReadeP — CHANGES_REQUESTED** [review]
Looks great Richie! Awesome to see you making so much progress this semester on Scoop. 

I left you some feedback, please address it and request me for a second review before merging.

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (rs929) [reply Scoop/Controller/Onboarding/OnboardingViewController.swift]: I tried making it private, but it ended up being inaccessible to the rest of the view controllers, so I'm not sure I could set it private

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (rs929) [reply Scoop/Controller/Onboarding/OnboardingViewController.swift:19]: I tried making it private, but it ended up being inaccessible to the rest of the view controllers, so I'm not sure I could set it private

File: Scoop/Controller/Onboarding/PreferencesViewController.swift
Comment (emarc0314) [inline Scoop/Controller/Onboarding/PreferencesViewController.swift:67]: remove top and bottom anchor

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (emarc0314) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift]: remove top anchor

File: Scoop/Controller/Onboarding/ProfilePictureViewController.swift
Comment (emarc0314) [inline Scoop/Controller/Onboarding/ProfilePictureViewController.swift:108]: remove top constraints from uploadButton

File: Scoop/Controller/Onboarding/ProfilePictureViewController.swift
Comment (emarc0314) [inline Scoop/Controller/Onboarding/ProfilePictureViewController.swift:121]: remove bottom constraints from profileButton

**Review by emarc0314 — CHANGES_REQUESTED** [review]

**Review by emarc0314 — APPROVED** [review]
noice 🔥

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:15]: nit: might want to alphabetize these `private let`s

File: Scoop/Controller/Onboarding/AboutYouViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift]: If these colors are reused elsewhere, for example in Eatery the `eatery blue` color is reused a lot, you might want to consider setting up an extension to UIColor and access these color as a static property

File: Scoop/Controller/Onboarding/FavoritesViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:15]: Same as above

File: Scoop/Controller/Onboarding/FavoritesViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift]: Same as above

File: Scoop/Controller/Onboarding/OnboardingViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift]: I don't think this needs to be a label. If you aren't adding any text, you can just use a `UIView`

File: Scoop/Controller/Onboarding/PhoneNumberViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/PhoneNumberViewController.swift]: Same as above

File: Scoop/Controller/Onboarding/PhoneNumberViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/PhoneNumberViewController.swift:60]: You might need to add an `accessibilityLabel` here. Since if someone is using the app with a voice assistant, text in all caps will be read out character by character, as opposed to just the text.

File: Scoop/Controller/Onboarding/PreferencesViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/PreferencesViewController.swift:61]: Same here.

File: Scoop/Controller/Onboarding/PreferencesViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/PreferencesViewController.swift:82]: Same as above.

File: Scoop/Controller/Onboarding/PreferredContactViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/PreferredContactViewController.swift:63]: Same as above

File: Scoop/Controller/Onboarding/ProfilePictureViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/ProfilePictureViewController.swift]: Instead of adding spaces to the title string, you might want to setup `insets` instead.

File: Scoop/Controller/Onboarding/ProfilePictureViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/ProfilePictureViewController.swift]: Same as above.

File: Scoop/Controller/Onboarding/ProfilePictureViewController.swift
Comment (NgaiJustin) [inline Scoop/Controller/Onboarding/ProfilePictureViewController.swift]: Same as above.

**Review by NgaiJustin — CHANGES_REQUESTED** [review]
Great work! Let some small comments

**Review by NgaiJustin — APPROVED** [review]
👍

**Review by ReadeP — APPROVED** [review]
Looks great Richie! Awesome work

**Pull request merged**
